### DISCUSSION TITLE
error: rate-overlimit

### DISCUSSION THREAD

**Discussion by bruno918** [WhiskeySockets/Baileys#395] [General Questions] [answered]
Recently I started to come across the following error, rate-overlimit, it is caused when many commands are sent in a row... This error is new to me and is directed to the library.
Can anyone tell me something? I'm going to make an "antispam" that solves part of the problem, but that certainly couldn't happen, right?

**Comment by Fonnte** [comment]
You are reading right, rate overlimit. So try send message slower should work.

**Comment by bruno918** [comment]
I thought it was an error in my code or something, as I've never come across this before... so is it Baileys that doesn't support many messages? In this case, there aren't many alternatives, right?

**Reply by bruno918** [reply]
my bot receives a lot of requests from many groups, so I was looking for a way around this.

**Comment by bruno918** [comment]
The problem is with Baileys, I tested a completely different code and the same thing happened

**Comment by okfull** [comment, marked as answer]
I work with bots in more than 100 groups using baileys, this code helped me to no longer have rate-overlimit

```
const groupCache = new NodeCache()

async function getGroupMetaData(from){
	if (!from?.endsWith(`g.us`)){
		return {}
	}

	var metadata = groupCache.get(from)

	if (!metadata?.subject){
		metadata = await sock.groupMetadata(from).catch(() => null)

		metadata = {
			owner: metadata?.owner || metadata?.participants?.filter(i => i.admin == 'superadmin')[0]?.id,
			participants: metadata?.participants || [],
			subject: metadata?.subject,
			restrict: metadata?.restrict,
			announce: metadata?.announce,
		}
 
		groupCache.set(from, metadata)
	}

	return metadata
}

async function reply(jid, message, options){
	const sent = await sock.sendMessage(jid, {
		text: message
	}, {
		ephemeralExpiration: options?.ephemeralExpiration,
		quoted: options.quoted,
		...(isJidGroup(jid) ? {
			cachedGroupMetadata: async(jid) => {
				return groupCache.get(jid)
			}
		} : {})
	})

	return sent
}
```

**Reply by bruno918** [reply]
Thank you, I will test, know that you are saving me

**Comment by bruno918** [comment]
One way to solve it was to simply delete the code for this part in Baileys, my code hardly crashes now.
